OpenRA.Mods.Common/Widgets/ProductionPaletteWidget.cs

```diff
@@ -577,9 +579,26 @@ public override void Draw()
 							icon.Pos + infiniteOffset,
 							Color.White, Color.Black, 1);
 					else if (total > 1 || waiting)
+					{
+						var size = overlayFont.Measure(total.ToString());
+						float2 pos;
+						switch (QueuedTextAlign)
+						{
+							case TextAlign.Center:
+								pos = new float2(QueuedOffset.X - size.X / 2, QueuedOffset.Y);
+								break;
+							case TextAlign.Right:
+								pos = new float2(QueuedOffset.X - size.X, QueuedOffset.Y);
+								break;
+							default:
+								pos = QueuedOffset;
+								break;
+						}
```

Comment:
Can't these be bakes into the offset?

Reply:
Do you mean removing this altogether and just adjusting queued offset from `ChromeMetrics`? Possible, but depending on the queued count the number could overflow the icon bounds. I thought about it, but I think it's better to do this properly (if I'm already changing this code).

Reply:
Ah, I understand. This should be fine. The only tiny issue I can see is that it measures the size even when it's unused

Reply:
So something like this?

```
float2 pos = QueuedOffset;
if (QueuedTextAlign != TextAlign.Left)
{
    var size = overlayFont.Measure(total.ToString());

    switch (QueuedTextAlign)
    {
        case TextAlign.Center:
            pos = new float2(QueuedOffset.X - size.X / 2, QueuedOffset.Y);
            break;
        case TextAlign.Right:
            pos = new float2(QueuedOffset.X - size.X, QueuedOffset.Y);
            break;
    }
}

```

Or without `switch`?

```
float2 pos = QueuedOffset;
if (QueuedTextAlign != TextAlign.Left)
{
    var size = overlayFont.Measure(total.ToString());
    
    pos = QueuedTextAlign == TextAlign.Center ?
        new float2(QueuedOffset.X - size.X / 2, QueuedOffset.Y) :
        new float2(QueuedOffset.X - size.X, QueuedOffset.Y);
}
```

Reply:
It's cleaner without the switch

Reply:
`float2 pos = QueuedOffset;` -> `var pos = QueuedOffset;` then this should be good to go.